Source/JavaScriptCore/heap/StructureAlignedMemoryAllocator.cpp

```diff
@@ -164,6 +164,12 @@ void StructureAlignedMemoryAllocator::initializeStructureAddressSpace()
 
 void* StructureAlignedMemoryAllocator::tryMallocBlock()
 {
+#if PLATFORM(PLAYSTATION)
+    static std::once_flag s_onceFlag;
+    std::call_once(s_onceFlag, [] {
+        initializeStructureAddressSpace();
```

Comment:
How does this work? Does Playstation not freeze the config page or something?

Reply:
No, it is working fine. I'm not sure your warry. What is the issue?

Reply:
We call `permanentlyFreeze()` on our configuration when the VM constructor is called, which `mmaps` the configuration to be non-writable. It looks like we call `permanentlyFreeze()` at the end of the VM constructor though, so we will have already created structures at that point.

That said, I think we should only have one place we initialize the structure address space. Can we just do it in tryMallocBlock?

Reply:
Okay, I'll resend the patch.